### ISSUE TITLE
PyMC 3.7: plot_posterior `ref_val` argument not backwards compatible

### ISSUE THREAD

**Issue by clausherther** [pymc-devs/pymc#3498]
In <= 3.6, I was able to pass in a list of values for each variable to the `ref_val` argument in plot_posterior, such as
```
[0.5012363643372008,
 0.4182023273414778,
 -0.08303403699572298,
 0.834341554397153]
```
```
with model:
    pm.plot_posterior(trace,  varnames=["coeff",  "coeff_diff", "coeff_perc_diff"], ref_val=ref_vals)
```
In 3.7, this results in an error:
```
ValueError: Argument `ref_val` must be None, a constant, or a dictionary like {"var_name": {"ref_val": (lo, hi)}}

```
This is because the new arviz `plot_posterior` takes a different format for the `ref_val` argument. 
Also, there doesn't seem to be a good example of how to reformat code to support this, especially with multivalued variables, such as multiple coefficients. 
I'd be happy to update the doc string or doc website, but I can't seem to find any good hints on how to format the reference values correctly.

**Comment by clausherther** [comment]
I have tried passing this dictionary:
```
ref_vals_dict = {'coeff': {'ref_val': 0.4182023273414778},
 'coeff_diff': {'ref_val': -0.08303403699572298},
 'coeff_perc_diff': {'ref_val': 0.834341554397153}}
```
to:
`pm.plot_posterior(trace,  kind="hist", var_names=var_names, ref_val=ref_vals_dict, round_to=2)`

And get this error:
```
~/.pyenv/versions/3.6.7/lib/python3.6/site-packages/arviz/plots/posteriorplot.py in display_ref_val()
    219             for sel in ref_val.get(var_name, []):
    220                 if all(
--> 221                     k in selection and selection[k] == v for k, v in sel.items() if k != "ref_val"
    222                 ):
    223                     val = sel["ref_val"]

AttributeError: 'str' object has no attribute 'items'
```
What am I missing?

Btw, the docs on this still say
```
ref_val: float or dictionary of floats

    display the percentage below and above the values in ref_val. If a list is provided, its length should match the number of variables.
```

**Comment by martintburgess** [comment]
Any update on this? Having exactly the same issue trying to plot reference values.

**Comment by clausherther** [comment]
This is still one of my biggest frustrations with using arviz instead of the old PyMC3 plots. Any thoughts on how to correctly pass the ref_vals? There does not seem to be documentation around what the plot_posterior `ref_vals` argument expects.

**Comment by aloctavodia** [comment]
fixed by arviz-devs/arviz#830

**Comment by clausherther** [comment]
That is so exciting! Thanks @aloctavodia!
